### PR TITLE
Update options for ingress to represent state-of-the-art

### PR DESCRIPTION
**Pull request by christianknell**
Refactor ingress route

- best practices
- For simplification serviceName and servicePort have been defined as variables.

### CODE REVIEW

**Comment by christianknell** [comment]
Do you need any more information on this one @EugenMayer?

**Comment by EugenMayer** [comment]
Hello Chris, no i do not think.

But also, the level of individual settings in the ingress you are trying to go for is the reason why there is the way to disable the ingress entry - and create a new one. So i rather tend to not overcomplicate the chart for the sake of entirely flexibility.

That does not mean to lock anybody out, but disabling the ingress entry and create your own is easy using a helm wrapping chart or more suffisticated terraform - that is the best practice for ingress due to the massive amount of configuration that can go in and need to be wrapped.

What are your thoughts?

**Comment by EugenMayer** [comment]
i reviewed your branch once again and happen to notice that your partially did the opposite what i was thinking, you actually made the ingress rs 1.19+ compatible .. i was not aware it was that outdated since i was not using it at all.

I cherry picked that one just due to it's urgency change in https://github.com/EugenMayer/helm-charts/commit/8762e78b1d340d759d0dae7030cb03318894ee18 - i hope you do not mind. I like the serviceName/port functions, we could go with those too. 

If you could strip down your PR to:
 - only 1.19+compat
 - no host or path mappings (those are subject for custom ingress definitions IMHO)

So basically the helper methods, i would love to merge that one. Thank you for your effort!

**Comment by christianknell** [comment]
Hey @EugenMayer,

i don't think it overcomplicates the HelmChart at all. The way i have defined the Ingress is also the official way you get when running `helm create my-chart`. With this definition it is now compatible throughout all version of Kubernetes and not only limited to 1.19+. But i see your point, that old versions of Kubernetes will hopefully be gone anyway someday. But at the moment i think a good Helm Chart should still support them, because they are still used.

What do you mean with host and path mapping? The only difference in my MR is the way you define the path. Because this is a per-host setting in each ingress definition, it should also be defined for each host individual. If you only have one host, then the number of values you have to define doesn't differ from the current way of implementation. But it gives you more flexibility if you need to expose it in another way. Using a Helm Wrapping chart or Terraform is a overcomplication from my point of view and it is also a bad fit, when using this Helm Chart with GitOps tools like Flux or ArgoCD.

Also the current way of defining the paths doesn't work at all. I will open a Issue to explain their a bit more.

The way you cherry-picked the serviceName and servicePort doesn't work by the way, because you forgot the definition in the _helpers.tpl-file. So that should be added again.

**Review by EugenMayer — CHANGES_REQUESTED** [review]
Please remove all sections to support < 1.19 - not going to support EOL versions of k8s

https://endoflife.date/kubernetes

**Review by EugenMayer — APPROVED** [review]
LGTM, thank you a lot for your effort!
